hr_expense_product_policy/models/hr_employee.py

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
+# © 2016 OpenSynergy Indonesia
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import fields, models, api
+
+
+class HrEmployee(models.Model):
+    _inherit = "hr.employee"
+
+    @api.depends(
+        "job_id.all_allowed_expense_product_ids",
+        "job_id.all_allowed_expense_product_ids.hr_expense_ok",
+        "job_id",
+        "department_id.all_allowed_expense_product_ids",
+        "department_id.all_allowed_expense_product_ids.hr_expense_ok",
+        "allowed_expense_product_categ_ids",
+        "allowed_expense_product_ids",
+        "allowed_expense_product_ids.hr_expense_ok"
+    )
+    def _compute_all_allowed_product_ids(self):
+        obj_product = self.env["product.product"]
+        for employee in self:
+            products = employee.allowed_expense_product_ids
+            category_ids = employee.allowed_expense_product_categ_ids.mapped(
+                "id")
+            criteria = [
+                ("categ_id", "in", category_ids),
+                ("hr_expense_ok", "=", True),
+                ]
+            products += obj_product.search(criteria)
+            if employee.department_id:
+                department = employee.department_id
+                products += department.all_allowed_expense_product_ids
+
+            if employee.job_id:
+                job = employee.job_id
+                products += job.all_allowed_expense_product_ids
+
+            employee.all_allowed_expense_product_ids = \
+                products.mapped("id")
```

Comment:
You are right again. Would you kindly tell me the different? Because the test did not fail when i use list of IDs. Why using actual recordset better? Thanks in advance.